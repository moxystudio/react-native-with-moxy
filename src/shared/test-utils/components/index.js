export { AppTree } from './app-tree';
