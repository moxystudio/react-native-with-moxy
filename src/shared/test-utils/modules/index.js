export { AppTree } from './react-app-tree';
