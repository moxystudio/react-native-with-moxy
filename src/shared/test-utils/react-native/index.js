import { Platform } from 'react-native';

export const mockPlatform = (os) => {
    Platform.select.mockImplementation((platforms) => platforms[os]);
    Platform.OS = os;
};
