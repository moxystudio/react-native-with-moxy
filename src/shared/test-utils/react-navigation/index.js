/* eslint-env jest */

// Reference documentation: https://reactnavigation.org/docs/navigation-prop/

export const createNavigationProp = () => ({
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: () => ({ remove: jest.fn }),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    setOptions: jest.fn(),
    isFocused: jest.fn(() => false),
});
