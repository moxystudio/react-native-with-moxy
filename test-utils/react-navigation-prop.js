/* eslint-disable no-undef */
export default () => ({
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: () => ({ remove: jest.fn }),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn(() => false),
});
/* eslint-enable no-undef */

