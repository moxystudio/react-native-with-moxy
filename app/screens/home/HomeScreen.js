import React from 'react';
import { View, Text } from 'react-native';

import styles from './styles';

const HomeScreen = () => (
    <View style={ styles.container }>
        <Text>This is the Homescreen</Text>
    </View>
);

export default HomeScreen;
