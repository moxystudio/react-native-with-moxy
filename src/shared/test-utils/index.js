import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import idObj from 'identity-obj-proxy';
import { AppConfigProvider, ThemeProvider, LocaleProvider } from '../modules';

const localeConfig = {
    locales: [
        {
            id: 'en-US',
            name: 'English',
            messages: idObj,
        },
        {
            id: 'pt-PT',
            name: 'PortuguÃªs',
            messages: idObj,
        },
    ],
    // Consider the following React code:
    //
    // <Text>
    //    <FormattedMessage id="foo" />
    // </Text>
    //
    // In this case, the "foo" key will resolve into the text "bar", and the following will be rendered:
    //
    // <Text>
    //    <Text>
    //       bar
    //    </Text>
    // </Text>
    //
    // This a is necessary workaround because *ByText queries can only look up a string if it's
    // a direct child of a Text component.
    textComponent: Text,
};

const Providers = ({ children }) => (
    <AppConfigProvider>
        <SafeAreaProvider>
            <ThemeProvider>
                <LocaleProvider { ...localeConfig }>
                    { children }
                </LocaleProvider>
            </ThemeProvider>
        </SafeAreaProvider>
    </AppConfigProvider>
);

Providers.propTypes = {
    children: PropTypes.element.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: Providers, ...options });

// Export everything from NTL
export * from '@testing-library/react-native';

// Override render method
export { customRender as render };
