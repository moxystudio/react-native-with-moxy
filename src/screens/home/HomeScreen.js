import React, { useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import HomeHeader from './header';
import { Button, useTheme } from '../../shared/components';

import buildStyles from './styles';

const HomeScreen = ({ navigation }) => {
    navigation.setOptions({
        header: () => <HomeHeader />,
    });

    const { themeStyles } = useTheme();
    const styles = useMemo(() => buildStyles(themeStyles), [themeStyles]);

    const handlePress = useCallback(() => {
        navigation.navigate('Profile', { screen: 'Profile2' });
    }, [navigation]);

    return (
        <View style={ styles.container }>
            <Text style={ styles.text }>This is the Homescreen</Text>
            <Button
                type="highlight"
                title="Go To Profile"
                style={ styles.button }
                onPress={ handlePress }
                textStyle={ styles.buttonText }
                underlayColor={ themeStyles.colors.terciary } />
        </View>
    );
};

HomeScreen.propTypes = {
    navigation: PropTypes.object.isRequired,
};

export default HomeScreen;
