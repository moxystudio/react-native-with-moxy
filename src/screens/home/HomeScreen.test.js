import React from 'react';
import { mount } from 'enzyme';
import HomeScreen from './HomeScreen';
import { AppTree } from '../../shared/test-utils/components';
import createNavigation from '../../shared/test-utils/react-navigation-prop';

const navigation = createNavigation();

beforeEach(() => {
    navigation.navigate.mockClear();
});

it('should render correctly', () => {
    const tree = mount(
        <AppTree>
            <HomeScreen navigation={ navigation } />
        </AppTree>,
    );

    const Button = tree.find('Button');

    Button.props().onPress();

    expect(tree).toMatchSnapshot();
    expect(navigation.navigate).toHaveBeenCalled();
});
