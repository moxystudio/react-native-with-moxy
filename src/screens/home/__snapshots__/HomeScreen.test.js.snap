// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<AppTree>
  <ThemeContextProvider>
    <HomeScreen
      navigation={
        Object {
          "addListener": [Function],
          "closeDrawer": [MockFunction],
          "dismiss": [MockFunction],
          "dispatch": [MockFunction],
          "getParam": [MockFunction],
          "goBack": [MockFunction],
          "isFocused": [MockFunction],
          "navigate": [MockFunction] {
            "calls": Array [
              Array [
                "Profile",
                Object {
                  "screen": "Profile2",
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "openDrawer": [MockFunction],
          "pop": [MockFunction],
          "popToTop": [MockFunction],
          "push": [MockFunction],
          "replace": [MockFunction],
          "setOptions": [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "header": [Function],
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "setParams": [MockFunction],
          "state": Object {
            "params": Object {},
          },
          "toggleDrawer": [MockFunction],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#221240",
                "fontFamily": "Lato-Regular",
                "fontWeight": "normal",
                "paddingBottom": 6,
                "paddingTop": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#221240",
                  "fontFamily": "Lato-Regular",
                  "fontWeight": "normal",
                  "paddingBottom": 6,
                  "paddingTop": 4,
                }
              }
            >
              This is the Homescreen
            </Text>
          </Text>
          <Button
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#221240",
                "borderRadius": 20,
                "height": 50,
                "justifyContent": "center",
                "marginTop": 20,
                "width": 200,
              }
            }
            textStyle={
              Object {
                "color": "#2932D3",
                "fontFamily": "Lato-Bold",
                "fontWeight": "normal",
                "paddingBottom": 6,
                "paddingTop": 4,
              }
            }
            title="Go To Profile"
            type="highlight"
            underlayColor="#F64F51"
          >
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#221240",
                  "borderRadius": 20,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 200,
                }
              }
              underlayColor="#F64F51"
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#221240",
                    "borderRadius": 20,
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 20,
                    "width": 200,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#221240",
                      "borderRadius": 20,
                      "height": 50,
                      "justifyContent": "center",
                      "marginTop": 20,
                      "width": 200,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#2932D3",
                        "fontFamily": "Lato-Bold",
                        "fontWeight": "normal",
                        "paddingBottom": 6,
                        "paddingTop": 4,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#2932D3",
                          "fontFamily": "Lato-Bold",
                          "fontWeight": "normal",
                          "paddingBottom": 6,
                          "paddingTop": 4,
                        }
                      }
                    >
                      Go To Profile
                    </Text>
                  </Text>
                </View>
              </View>
            </TouchableHighlight>
          </Button>
        </View>
      </View>
    </HomeScreen>
  </ThemeContextProvider>
</AppTree>
`;
